lib/permutations.rb

```diff
@@ -1,4 +1,21 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented" 
+
+  return false if string1.length != string2.length
+
+  string1_array = string1.chars
+  string2_array = string2.chars
+
+  hash = {}
+  string1_array.each do |element|
+    hash[element] = string1_array.count(element)
+  end
```

Comment:
See above how `.count` is an O(n) method.  This solution is an O(n^2) solution

Consider counting up like I suggested above.